api/tests/unit/integrations/github/test_unit_github_views.py

```diff
@@ -556,17 +557,62 @@ def test_fetch_issues_returns_error_on_bad_response_from_github(
     mocker: MockerFixture,
 ) -> None:
     # Given
-    mocker.patch("requests.get", side_effect=mocked_requests_get_error)
+
+    mock_response = {
+        "message": "Validation Failed",
+        "errors": [{"message": "Error", "code": "not_found"}],
+        "documentation_url": "https://docs.github.com/v3/search/",
+        "status": "404",
+    }
+
+    responses.add(
+        method="GET",
+        url="https://api.github.com/search/issues?q=%20repo:repo/repo%20is:issue%20is:open%20in:title%20in:body&per_page=100&page=1",  # noqa: E501
+        status=status.HTTP_404_NOT_FOUND,
+        json=mock_response,
+    )
     url = reverse("api-v1:organisations:get-github-issues", args=[organisation.id])
     data = {"repo_owner": "owner", "repo_name": "repo"}
     # When
     response = admin_client_new.get(url, data=data)
 
     # Then
     assert response.status_code == status.HTTP_502_BAD_GATEWAY
+    assert "Failed to retrieve GitHub issues." in response.json()["detail"]
+
+
+@responses.activate
+def test_search_issues_returns_error_on_bad_search_params(
+    admin_client_new: APIClient,
+    organisation: Organisation,
+    github_configuration: GithubConfiguration,
+    github_repository: GithubRepository,
```

Comment:
Nit:

```suggestion
    github_configuration: GitHubConfiguration,
    github_repository: GitHubRepository,
```

Reply:
I can't change this because it is the model's name.
https://github.com/Flagsmith/flagsmith/blob/135711662f938fa7a2518ac2f7a1c70ac75b0cc3/api/integrations/github/models.py#L19

Reply:
Ok I guess it's too much work to change the name of the model?

Reply:
I decided to rename the model, thanks for the suggestion. I will merge this PR afterward.